c/src/lib/libbsp/arm/shared/stm32fxxxx/error/hal-fatal-error-handler.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file hal-fatal-error-handler.c
+ *
+ * @ingroup error
+ *
+ * @brief Error handler for CPU exceptions
+ *
+ */
+
+/*
+ * Copyright (c) 2015 Vecna Technologies, Inc.
+ *
+ * The license and distribution terms for this file may be
+ * found in the file LICENSE in this distribution or at
+ * http://www.rtems.com/license/LICENSE.
+ */
+
+#include <rtems.h>
+#include <rtems/score/userextimpl.h>
+#include <rtems/score/cpu.h>
+
+#include <hal-fatal-error-handler.h>
+
+
+typedef struct
+{
+	uint64_t reg_value;
+}register_double;
+
+void stm32fxxxx_fatal_error_handler(Internal_errors_Source source, bool is_internal, CPU_Exception_frame *frame)
+{
+	register_double *fpu_reg;
+
+	int *ptr = (int *)frame;
+	int count=0;
+
+	printf ("\n --- Hard Fault Occurred!!! :( --- \n");
+	printf ("\n RTEMS Error Code : %d \n", source);
+	printf (" Caused by RTEMS?: ");
+	printf (is_internal == false ? "Nope\n" : "Yup\n");
+	printf (" Hard fault caused by vector # %x\n", frame->vector);
+	printf ("\n Providing Stacked Register Details... \n");
+	printf ("\n General Purpose Registers:\n");
+
+	for(count=0;count<=12;count++)
+	{
```

Comment:
These are the registers we're printing out?  Why aren't we printing all 31 variable ones then?